src/binary/raw_binary_writer.rs

```diff
@@ -777,6 +777,10 @@ impl<'a, W: Write> Writer for RawBinaryWriter<W> {
             Ok(())
         })?;
 
+        // Now that we know how large the container's header is, add its length to the
+        // calculate container size.
+        let container_size = container_size + header_io_range.len();
```

Comment:
🗺️ Prior to this patch, the binary reader would ignore the annotations wrappers' combined length fields as they were largely redundant. It turns out the binary writer had been calculating that field incorrectly; because the `ion-rust` reader was usually (always?) the one consuming the `ion-rust` binary writer's output, the problem went unnoticed.